### PR TITLE
Implement Auto Craft Ingredients option

### PR DESCRIPTION
**Pull request by MassCraxx**
This should add an new TickBox option in the UI tab to let the AutoCook consider ingredients from closed cans in the vicinity.
Closed cans chosen for the recipe and all required tools will be taken into the inventory, used to craft the usable opened ingredient and then finally used in the AutoCook process.

Open:
- <s>Recipe code is unstable and will get a rework from the AutoEat mod</s>
- <s>Context menu option will not show up if there are only valid closed cans around</s>
- <s>UI</s>
- <s>optimizations, caching, reduce load</s>

Known Issues:
- if there is only a craftable valid spice in the vicinity (e.g Gravy Mix in Roasting Pan), autocook will be available, but simply craft the spice and stop doing anything, as the recipe can not be started with a spice... Looking for a way to filter it out pre-craft without creating a fake item..
-- turns out gravy is the only vanilla case where this happens. Excluding this one seems good enough

- <s>choosing algorithm prefers crafting items, resulting in more crafts than uses if available</s>
- <s>as long as there are possible craftable ingredients around, autocook will not stop until its all crafted</s>

### CODE REVIEW

File: AutoCook/media/lua/client/AutoCook.lua [hunk 1]
```diff
@@ -396,3 +495,6 @@ function AutoCook:new(playerObj, recipe, baseItem)
     
     return o;
 end
+
+-- load persisted values on reload script
+AutoCook:init(getPlayer())
```
Comment (katupia) [inline AutoCook/media/lua/client/AutoCook.lua]: Keeping active debug code is only ok within a "false" condition by default. Maybe add some AutoCook.Debug same as there is an AutoCook.Verbose

Comment (MassCraxx) [reply AutoCook/media/lua/client/AutoCook.lua]: agreed, will only happen in debug mode now

**Comment by MassCraxx** [comment]
Reworked this feature to work with all crafted food ingredients, not just cans.
Also reworked ContextMenu for it to show up under new conditions. Has tooltip now:
image
image

added UI for AutoCraft
image

**Comment by MassCraxx** [comment]
Changed icon to show result.
image

**Pull request merged**